cmake/MiNiFiOptions.cmake

```diff
@@ -42,6 +42,7 @@ add_minifi_option(DOCKER_SKIP_TESTS "Skip building tests in docker image targets
 add_minifi_option(DOCKER_PUSH "Push created images to the specified tags" OFF)
 
 add_minifi_option(PORTABLE "Instructs the compiler to remove architecture specific optimizations" ON)
+add_minifi_option(MINIFI_USE_CONAN_PACKAGER "Instructs cmake to use conan packager to install external libs & build MiNiFi C++" OFF)
```

Comment:
Instead of a single option, I think setting the package source per dependency would be better. I'm thinking about new CMake options like `MINIFI_LIBCURL_SOURCE` with possible values like `system` (find_package), `conan` or `build` (ExternalProject, FetchContent, etc). If you prefer not adding `system` now, that's fine.

You can see an example option with a predetermined set of possible values on line 76-78.

Reply:
@szaszm I think I see where you are talking about with a custom option that has multiple possible values. This is the example I found in MiNiFiOptions.cmake

~~~cmake
# Option: STRICT_GSL_CHECKS
# AUDIT: Enable all checks, including gsl_ExpectsAudit() and gsl_EnsuresAudit()
# ON: Enable all checks, excluding gsl_ExpectsAudit() and gsl_EnsuresAudit() (GSL default)
# DEBUG_ONLY: Like ON in the Debug configuration, OFF in others (MiNiFi C++ default)
# OFF: Throw on contract checking and assertion failures instead of calling std::terminate()
set(STRICT_GSL_CHECKS "DEBUG_ONLY" CACHE STRING "Contract checking and assertion failures call terminate")
list(APPEND STRICT_GSL_CHECKS_Values AUDIT ON DEBUG_ONLY OFF)
set_property(CACHE STRICT_GSL_CHECKS PROPERTY STRINGS ${STRICT_GSL_CHECKS_Values})
~~~

I agree I think it would be great to create new the CMake options for the source of how we retrieve external libs like the following:

~~~cmake
# Option: MINIFI_LIBCURL_SOURCE
# BUILD: MiNiFi LibCURL source retrieved from CMake's ExternalProject, FetchContent, etc
# CONAN: MiNiFi LibCURL source retrieved from conan packager
# SYSTEM: MiNiFi LibCURL source retrieved from native OS system
set(MINIFI_LIBCURL_SOURCE "BUILD" CACHE STRING "Builds LibCURL from source using standalone CMake approach"
list(APPEND MINIFI_LIBCURL_SOURCE_Values SYSTEM BUILD CONAN)
set_property(MINIFI_LIBCURL_SOURCE PROPERTY STRINGS ${MINIFI_LIBCURL_SOURCE_Values})
~~~

Reply:
I will make the update to this PR for **OpenSSL, LibCURL, ZLib** @szaszm

Reply:
Currently in our conanfile.py, we have the following MINIFI external libraries sources set to CONAN:

~~~python
        tc.variables["MINIFI_LIBCURL_SOURCE"] = "CONAN"
        tc.variables["MINIFI_OPENSSL_SOURCE"] = "CONAN"
        tc.variables["MINIFI_ZLIB_SOURCE"] = "CONAN"
        tc.variables["MINIFI_CIVETWEB_SOURCE"] = "CONAN"
        tc.variables["MINIFI_LIBXML2_SOURCE"] = "CONAN"
        tc.variables["MINIFI_FMT_SOURCE"] = "CONAN"
        tc.variables["MINIFI_SPDLOG_SOURCE"] = "CONAN"
        tc.variables["MINIFI_CATCH2_SOURCE"] = "CONAN"
~~~